--- db-daemon.c.orig	2019-07-31 15:15:37.929234086 -0700
+++ db-daemon.c	2019-07-31 15:14:55.002234086 -0700
@@ -28,8 +28,9 @@
  * Archlinux also). But if someone needs no systemd in dependencies,
  * it can be easily turned off, check the code in main() - conditions on
  * getenv("NOTIFY_SOCKET").
+   crux doesn't use systemd; removing this reference
+   #include <systemd/sd-daemon.h>
  */
-#include <systemd/sd-daemon.h>
 #else // !WIN32
 // parameters for a client pipe thread
 struct thread_param {
@@ -787,13 +788,10 @@
 
     /* now ready for serving requests, notify parent */
     /* FIXME: OS dependent code */
-    if (getenv("NOTIFY_SOCKET")) {
-        sd_notify(1, "READY=1");
-    } else {
-        if (write(ready_pipe[1], "ready", strlen("ready")) != strlen("ready"))
-            perror("failed to notify parent");
-        close(ready_pipe[1]);
-    }
+    /* Removing notify_socket decision path */
+    if (write(ready_pipe[1], "ready", strlen("ready")) != strlen("ready"))
+        perror("failed to notify parent");
+    close(ready_pipe[1]);
 
     create_pidfile(&d);
 
