--- db-daemon.c	2021-07-10 03:13:14.000000000 +0000
+++ db-daemon.c.new	2021-08-18 07:29:42.929000000 +0000
@@ -23,17 +23,11 @@
 #include <vchan-common.h>
 #endif
 
-#ifndef WIN32
-#ifdef HAVE_SYSTEMD
-#include <systemd/sd-daemon.h>
-#endif
-#else // !WIN32
 // parameters for a client pipe thread
 struct thread_param {
     struct db_daemon_data *daemon;
-    LONGLONG id;
+    long id;
 };
-#endif
 
 #include <qubesdb.h>
 #include "qubesdb_internal.h"
@@ -805,11 +799,6 @@
 
     /* now ready for serving requests, notify parent */
     /* FIXME: OS dependent code */
-#ifdef HAVE_SYSTEMD
-    if (getenv("NOTIFY_SOCKET")) {
-        sd_notify(1, "READY=1");
-    } else
-#endif /* HAVE_SYSTEMD */
     {
         if (write(ready_pipe[1], "ready", strlen("ready")) != strlen("ready"))
             perror("failed to notify parent");
