--- /root/saved/setup-ip        2019-06-20 15:49:41.612000000 -0700
+++ /usr/lib/qubes/setup-ip     2019-06-21 08:07:22.371000000 -0700
@@ -90,14 +90,15 @@
             /sbin/ifconfig "$INTERFACE" add "$ip6"/128
         fi
         /sbin/ifconfig "$INTERFACE" up
-        /sbin/route add -host "$gateway" dev "$INTERFACE"
+        /sbin/ip route add "$gateway"/32 dev "$INTERFACE"
+       /sbin/ip route add default via "$gateway"
         if [ -n "$gateway6" ] && ! echo "$gateway6" | grep -q "^fe80:"; then
-            /sbin/route -6 add "$gateway6/128" dev "$INTERFACE"
+            /sbin/ip route -6 add "$gateway6/128" dev "$INTERFACE"
         fi
         if ! qsvc disable-default-route ; then
-            /sbin/route add default gw "$gateway"
+            /sbin/ip route add default gw "$gateway"
             if [ -n "$gateway6" ]; then
-                /sbin/route -6 add default gw "$gateway6" dev "$INTERFACE"
+                /sbin/ip route -6 add default gw "$gateway6" dev "$INTERFACE"
             fi
         fi
         if ! is_protected_file /etc/resolv.conf ; then
